src/server_manager/model/account.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {CloudProvider} from './cloud';
+
+export interface Account {
```

Comment:
This abstraction seems off. Different clouds may have different credentials, so it's odd that they are sharing the same credentials implementation.

I think we need to define a CloudAccount interface, with id() and displayName() methods, and then have subclasses, like DigitalOceanAccount that implements that.

Later, you can have cloud providers that take the account. The GoogleCloudProvider would take the GoogleCloudAccount, but not a DigitalOceanAccount. Those things need to have different types.

Reply:
We need to think of the user journey of creating a server.

Do you pick the cloud provider, then the account? Or do you pick the account first?

Perhaps the account should have hide the cloud provider and provide a createServer() method